src/main/java/ch/njol/util/VectorMath.java

```diff
@@ -18,184 +18,84 @@
  */
 package ch.njol.util;
 
+import ch.njol.skript.effects.EffVectorRotateAroundAnother;
+import ch.njol.skript.effects.EffVectorRotateXYZ;
+import ch.njol.skript.expressions.ExprVectorCylindrical;
+import ch.njol.skript.expressions.ExprVectorFromYawAndPitch;
+import ch.njol.skript.expressions.ExprVectorSpherical;
+import ch.njol.skript.expressions.ExprYawPitch;
 import org.bukkit.util.Vector;
+import org.jetbrains.annotations.ApiStatus;
 
-/**
- * @author bi0qaw
- */
-public class VectorMath {
+@Deprecated
+@ApiStatus.ScheduledForRemoval
+public final class VectorMath {
```

Comment:
Was there a reason to remove this class? It seems like there is now some duplicated code in the vector elements.

Reply:
This is from @kiip1 's PR

Reply:
Although a few duplicate cases have been introduced by these changes, I still feel moving the static methods to where they're used is beneficial, besides the content of these methods won't ever change so the duplication of the methods doesn't matter as much.

Another argument for this change is that there are many more cases where it isn't duplicated, so you'd either be left with unused methods in VectorMath (in case you keep all methods) or a very minimal VectorMath (in case you remove the duplicate methods). I prefer removing the entire utility class over keeping a partial class or having unused code.

At the end of the day you do have a fair argument, and there are multiple ways to solve the problem and I chose the solution which removes the entire class, at the cost of some duplicates.